keps/sig-apps/961-maxunavailable-for-statefulset/README.md

```diff
@@ -254,11 +264,144 @@ tried this feature in Alpha, we would have time to fix issues.
 - maxUnavailable greater than 1 with partition and maxUnavailable greater than replicas
 
+## Test Plan
+For `Alpha`, unit tests and e2e tests will be added to test functionality at both
+with feature flag enabled and disabled. Defaults will be verified so that users
+who donot set this flag are not surprised at all.
+
+
 ## Graduation Criteria
 
-- Alpha: Initial support for maxUnavailable in StatefulSets added. Disabled by default. 
-- Beta:  Enabled by default with default value of 1. 
+- Alpha: Initial support for maxUnavailable in StatefulSets added. Disabled by default with default value of 1.
+- Beta:  Enabled by default with default value of 1 with upgrade downgrade testedd at least manually.
+
+
+## Production Readiness Review Questionnaire
+
+### Feature Enablement and Rollback
+
+###### How can this feature be enabled / disabled in a live cluster?
+
+- [x] Feature gate (also fill in values in `kep.yaml`)
+  - Feature gate name: MaxUnavailableStatefulSet
+  - Components depending on the feature gate: kube-apiserver and kube-controller-manager
+
+###### Does enabling the feature change any default behavior?
+
+No, the default behavior remains the same.
+
+###### Can the feature be disabled once it has been enabled (i.e. can we roll back the enablement)?
+
+Yes this feature can be disabled. Once disabled, all existing  StatefulSet will
+revert to the old behavior where rolling update will proceed one pod at a time.
+
+###### What happens if we reenable the feature if it was previously rolled back?
+
+If the feature is enabled and then later disabled, we will revert the value of 
+maxUnavailable to the default value so that customers are not surprised.
+
+###### Are there any tests for feature enablement/disablement?
+yes, there are unit tests which make sure the field is correctly dropped 
+on feature enable and disabled
+
+### Rollout, Upgrade and Rollback Planning
+
+###### How can a rollout or rollback fail? Can it impact already running workloads?
+
+No, a rollout or rollback cannot fail.
+Yes, it can impact already running workloads.
+
+If a rolling update is in progress for a StatefulSet, while this feature is being enabled in kube-apiserver
+and kube-controller-manager, the StatefulSet controller can run into corner cases where it will take longer
+for the controller to converge. This will only happen if after enabling the feature, the customer also sets
+maxUnavailable to a number greater than 1,  but the invariants and the logic will ensure that there are never more than
+maxUnavailable pods with the same identity and never more than maxUnavailable being deleted.
+
+It can impact already running workloads if service owner accidentally sets maxUnavailable
+to a very high number and due to some bug, the StatefulSet pods are not coming up.
+
+###### What specific metrics should inform a rollback?
+The customer should look at StatefulSet.Status.Replicas to ensure if that number is always 
```

Comment:
This isn't a metric. An operator cannot track that for all statefulsets potentially across many different clusters.
We really need a metric for this.

Reply:
[This question is not strictly required for Beta - so you can change to TODO or sth and figure it out for beta.

Reply:
changed to TODO